server/src/modules/boards/boards.service.ts

```diff
@@ -107,4 +156,94 @@ export class BoardsService {
       throw new HttpException('Board not found', HttpStatus.NOT_FOUND);
     return existingBoard;
   }
+
+  async findBoardsOwnedByUser(
+    userId: string,
+    page: number,
+    limit: number,
+  ): Promise<SuperBoardDocument[]> {
+    if (limit == 0) return [];
+
+    const skip = (page - 1) * limit;
+    return this.boardModel
+      .find({ owner: userId })
+      .sort([['updatedAt', 'descending']])
+      .skip(skip)
+      .limit(limit)
+      .exec();
+  }
+
+  async countBoardsOwnedByUser(userId: string): Promise<number> {
+    return this.boardModel.countDocuments({ owner: userId }).exec();
+  }
+
+  async findBoardsSharedForUser(
+    userId: string,
+    page: number,
+    limit: number,
+  ): Promise<SuperBoardDocument[]> {
+    if (limit == 0) return [];
+
+    const skip = (page - 1) * limit;
+    return this.boardModel
```

Comment:
you could try to define the query and reuse it for countBoardsSharedForUser 
```
  const query = {
    $or: [
      { 'permissions.viewer': userId },
      { 'permissions.editor': userId },
      { 'permissions.moderator': userId },
    ],
  };

  const [total, boards] = await Promise.all([
    this.boardModel.countDocuments(query).exec(),
    this.boardModel
      .find(query)
      .sort([['updatedAt', 'descending']])
      .skip(skip)
      .limit(limit)
      .exec(),
  ]);

  return { total, boards };
```

Maybe it would be nice to define counting method that accepts the query and returns the count. Then, no need for 
countBoardsSharedForUser - it would me more generic ;)

Reply:
Also no need to select all the fields from the `board`. We could use `select` to define only needed fields

Reply:
I think we need all of them